File: ios/LocationSharingApp/LocationSharingApp/Domain/Repositories/AuthRepository.swift
import Foundation
import Combine

protocol AuthRepository {
    func signIn(email: String, password: String) async -> Result<AuthResult, AuthError>
    func signUp(email: String, password: String, displayName: String) async -> Result<AuthResult, AuthError>
    func signOut() async -> Result<Void, AuthError>
    func getCurrentUser() async -> User?
    func refreshToken() async -> Result<String, AuthError>
    func isUserAuthenticated() async -> Bool
    func resetPassword(email: String) async -> Result<Void, AuthError>
}

File: android/app/src/main/java/com/locationsharing/app/domain/repository/AuthRepository.kt
package com.locationsharing.app.domain.repository

import com.locationsharing.app.data.models.AuthResult
import com.locationsharing.app.data.models.User

interface AuthRepository {
    suspend fun signIn(email: String, password: String): Result<AuthResult>
    suspend fun signUp(email: String, password: String, displayName: String): Result<AuthResult>
    suspend fun signOut(): Result<Unit>
    suspend fun getCurrentUser(): User?
    suspend fun refreshToken(): Result<String>
    suspend fun isUserAuthenticated(): Boolean
    suspend fun resetPassword(email: String): Result<Unit>
}